SlevomatCodingStandard/Sniffs/Complexity/CognitiveSniff.php

```diff
@@ -136,22 +143,33 @@ public function process(File $phpcsFile, $stackPtr): void
 			return;
 		}
 
+		if ($this->maxComplexity !== null) {
+			// maxComplexity is deprecated... if set use it
+			$this->warningThreshold = $this->maxComplexity + 1;
+			$this->errorThreshold = $this->maxComplexity + 1;
+		}
+
 		$cognitiveComplexity = $this->computeForFunctionFromTokensAndPosition($stackPtr);
 
-		if ($cognitiveComplexity <= $this->maxComplexity) {
+		if ($cognitiveComplexity < $this->warningThreshold) {
 			return;
 		}
 
 		$name = $phpcsFile->getDeclarationName($stackPtr);
 
-		$phpcsFile->addError(
+		$method = $cognitiveComplexity >= $this->errorThreshold
+			? 'addError'
+			: 'addWarning';
+
+		call_user_func(
```

Comment:
Please use standard calls. You can use `...$errorParameters` as in some other sniffs.